controllers/src/main/java/io/syndesis/controllers/integration/online/ActivateHandler.java

```diff
@@ -67,29 +71,45 @@
         this.projectConverter = projectConverter;
     }
 
-    @Override
-    public Set<Integration.Status> getTriggerStatuses() {
-        return Collections.singleton(Integration.Status.Activated);
+    public Set<IntegrationState> getTriggerStatuses() {
+        return Collections.singleton(IntegrationState.Active);
     }
 
+
+    /**
+     *
+     * @param integration
+     * @param revision
+     * @return
+     */
     @Override
-    public StatusUpdate execute(Integration integration) {
+    public StatusUpdate execute(Integration integration, IntegrationRevision revision) {
+        //TODO: This is clearly wrong, we need something better than that.
```

Comment:
Might it make sense that we add a max or last version counter directly on the integration ?